contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -570,6 +627,33 @@ impl InterchainTokenService {
         Ok(())
     }
 
+    fn execute_link_token_message(
+        env: &Env,
+        LinkToken {
+            token_id,
+            token_manager_type,
+            source_token_address: _,
+            destination_token_address,
+            params: _,
+        }: LinkToken,
+    ) -> Result<(), ContractError> {
+        ensure!(
+            token_manager_type != TokenManagerType::NativeInterchainToken,
+            ContractError::InvalidTokenManagerType
```

Comment:
could perform this check in ABI encode/decode instead since it can never be used on the wire, in a separate PR